### ISSUE TITLE
[bug][2.0] CMakeToolchain does not respect os.sdk when targeting armv8 iOS simulator

### ISSUE THREAD

**Issue by theodelrieu** [conan-io/conan#10275]
Hello,

I'm trying to cross-build some iOS recipes, and I think there's a bug in CMakeToolchain. I use this profile:

```ini
[settings]
os=iOS
os.version=9.0
os.sdk=iphonesimulator
build_type=Release
arch=armv8
compiler=apple-clang
compiler.version=13.0
compiler.libcxx=libc++
compiler.cppstd=17
```

The `test_package` step fails with the following error:

```
ld: in /Users/theo/.conan2/p/f92ba1e43ba51502/p/lib/libsodium.a(libsodium_la-core.o), building for iOS, but linking in object file built for iOS Simulator, for architecture arm64
```

The generated `conan_toolchain.cmake` contains the following line:

>set(CMAKE_OSX_SYSROOT iphoneos CACHE STRING "" FORCE)

This does not happen when setting `arch=x86_64`, `CMAKE_OSX_SYSROOT` being correctly set to `iphonesimulator` in this case.

**Comment by memsharded** [comment]
Thanks for reporting, indeed it seems a bug. I think that it should be reproducible in 1.X, so I am targeting this for 1.45 (and it will be automatically merged to 2.0)

**Comment by memsharded** [comment]
Fixed in https://github.com/conan-io/conan/pull/10300, will be released in 1.45, feedback welcome!
